fx-src/FirefoxStorage.js

```diff
@@ -144,10 +148,10 @@ export default class FirefoxAdapter extends BaseAdapter {
   open() {
     const self = this;
     return Task.spawn(function* (){
-      const path = self._options.path || SQLITE_PATH;
-      const opts = { path, sharedMemoryCache: false };
       if (!self._connection) {
-        self._connection = yield Sqlite.openConnection(opts).then(self._init);
+        const path = self._options.path || SQLITE_PATH;
+        const opts = { path, sharedMemoryCache: false };
+        self._connection = yield Sqlite.openConnection(opts).then(FirefoxAdapter._init);
```

Comment:
Personally, I like "simple things should be simple; hard things should be possible" as a principle of API design, so in my perfect world, consumers wouldn't have to manage connections unless they had a reason to (as I do). That said, the existing API is not simple even in simple cases, since the consumer has to explicitly `open()` and `close()` the connections (because Promises). So maybe I will update the consumers.